### PR TITLE
mini-yetus: a yetus you fall in love with

### PR DESCRIPTION
**Pull request by shjala**
Add script to run yetus only on changes made in the current branch against the master branch. This is much faster than running yetus on the entire codebase. Now you can do `make mini-yetus`!

### CODE REVIEW

**Review by deitch — APPROVED** [review]
Yes, please!!

File: tools/mini-yetus.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+docker run --rm -v "$SRC_DIR":/src:delegated,z ghcr.io/apache/yetus:0.15.0 \
+    --basedir=/src \
+    --test-parallel=true \
+    --dirty-workspace \
+    --empty-patch \
+    --plugins=all \
+    --patch-dir=/src/yetus-output > /dev/null 2>&1
+
+echo "[+] Results:"
+cat "$SRC_DIR/yetus-output/results-full.txt"
```
Comment (rene) [inline tools/mini-yetus.sh:64]: IMHO we should let the users decide to which files they want to look at it and remove the output by themselves... in my case I go directly to the result file I'm interested, e.g., results-shellcheck.txt or results-golangcilint.txt, here it's enforced to always get the full results and remove the output dir, sometimes users can also want to keep the results saved....

File: tools/mini-yetus.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
```
Comment (rene) [inline tools/mini-yetus.sh:1]: I didn't see any "bashism" in the script to enforce bash... better to keep `/bin/sh`....

File: tools/mini-yetus.sh (hunk 1 above)
Comment (shjala) [reply tools/mini-yetus.sh:64]: sure.

**Comment by shjala** [comment]
btw the result is clickable if using something like VSCode.
image

File: tools/mini-yetus.sh (hunk 2 above)
Comment (shjala) [reply tools/mini-yetus.sh]: Are you bashing my script? may the `/bin/sh` be with you!

**Comment by OhmSpectator** [comment]
The CI version of Yetus tries its best to stop the new competitor:
```
tools/mini-yetus.sh:9:1:shellcheck: warning: Use var=$(command) to assign output (or quote to assign string). [SC2209]
tools/mini-yetus.sh:96:30:shellcheck: warning: In POSIX sh, &> is undefined. [SC3020]
```

**Comment by OhmSpectator** [comment]
I also have a suggestion: could we add the mini-yetus run as a Makefile target?

File: tools/mini-yetus.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
+#!/bin/sh
```
Comment (OhmSpectator) [inline tools/mini-yetus.sh:1]: SPDX header, please :)

Comment (shjala) [reply tools/mini-yetus.sh:1]: almost none of the files in tools have it, I though that's the tradition :D

Comment (OhmSpectator) [reply tools/mini-yetus.sh:1]: I don't think it was an intentional tradition =)

Comment (shjala) [reply tools/mini-yetus.sh:1]: 😃 done.

Comment (shjala) [reply tools/mini-yetus.sh:1]: oh no "Linux Foundation".

File: tools/mini-yetus.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        cp --parents -r "$PWD/$file" "$SRC_DIR/" || { echo "Failed to copy $file"; exit 1; }
+    else
+        echo "[!] File does not exist: $PWD/$file"
+        exit 1
+    fi
+done
+
+# copy all the dot files from root to the SRC_DIR, this includes all
+# the yetus configuration files.
+find . -maxdepth 1 -type f -name '.*' -exec cp --parents {} "$SRC_DIR/" \;
```
Comment (europaul) [inline tools/mini-yetus.sh:76]: nice! :heart:

File: Makefile [hunk 5]
```diff
@@ -459,6 +459,10 @@ yetus:
 		--plugins=all \
 		--patch-dir=/src/yetus-output
 
+mini-yetus:
```
Comment (OhmSpectator) [inline Makefile:462]: 🫶🏻🫶🏻🫶🏻

Comment (shjala) [reply Makefile:462]: 🫰🏼🫰🏼🫰🏼

File: tools/mini-yetus.sh [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+# check if we are in the root of the repository
+if [ ! -d .git ]; then
+    echo "[!] Error: This script must be run from the root of the EVE repository."
+    exit 1
+fi
+
+if [ -z "$SRC_BRANCH" ]; then
+    # No branch specified, use the main branch
+    SRC_BRANCH="master"
```
Comment (OhmSpectator) [inline tools/mini-yetus.sh:56]: Can we please use `orinig/master`?

File: tools/mini-yetus.sh [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+# Do you hate yetus like everyone else? well I can't do anything about that but
+# using this you can at least hate `make yetus` a little less. This script will
+# run yetus on the changes in your current branch compared to master (excluding
+# vendor changes) and output the results, much faster than running it on the
+# entire codebase with `make yetus`.
+
+FULL=false
+
+usage() {
+    echo "Usage: $0 [-b branch_name] [-f]"
```
Comment (OhmSpectator) [inline tools/mini-yetus.sh:14]: `-b` was replaced with `-s` and `-d` as far as I see

File: tools/mini-yetus.sh [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    echo "[+] No destination branch specified. Using the current branch: $DST_BRANCH"
+fi
+
+if ! git rev-parse --verify "$DST_BRANCH" >/dev/null 2>&1; then
+    echo "[!] Error: Branch '$DST_BRANCH' does not exist."
+    exit 1
+fi
+
+SRC_DIR=$(mktemp -d)
+git diff --name-only master.."$DST_BRANCH" | grep -v '/vendor/' | while IFS= read -r file; do
```
Comment (OhmSpectator) [inline tools/mini-yetus.sh:81]: I guess you mean:
```suggestion
git diff --name-only "$SRC_BRANCH".."$DST_BRANCH" | grep -v '/vendor/' | while IFS= read -r file; do
```

Comment (shjala) [reply tools/mini-yetus.sh:81]: good catch!

File: tools/mini-yetus.sh (hunk 7 above)
Comment (shjala) [reply tools/mini-yetus.sh]: fixed.

File: tools/mini-yetus.sh (hunk 8 above)
Comment (shjala) [reply tools/mini-yetus.sh]: fixed.

**Comment by OhmSpectator** [comment]
Ah, sorry for the failure of the new linter. Just rebase on the latest master

**Review by OhmSpectator — APPROVED** [review]
Looks great! You can also add info about the Maketarget into the commit message, but it's optional.

**Pull request merged**